### DISCUSSION TITLE
incomplete error message

### DISCUSSION THREAD

**Discussion by chris1278** [PHPMailer/PHPMailer#2955] [Q&A] [answered]
Hello, I have integrated the phpmailer for an extension of mine. This is a phpbb forum for which this extension is intended. Sending is not a problem, but if an error occurs it usually just shows this code here:

`SMTP connect() failed. https://github.com/PHPMailer/PHPMailer/wiki/Troubleshooting`

My phpmailer script looks like this:

```
$mail	= new PHPMailer();
$mail->isSMTP();
$mail->isHTML(true);
$mail->Username	= 'username';
$mail->Password	= 'userpassword';;
$mail->setFrom('sendmail', 'name');
$mail->Host	= 'smtp-host';;
$mail->Port	= 587;
$mail->SMTPAuth	= true;
$mail->setLanguage('de');
$mail->SMTPSecure = 'tls';
$mail->SMTPOptions = [
	'ssl' => [
		'verify_peer'			=> false,
		'verify_peer_name'		=> false,
		'allow_self_signed'		=> true,
	]
];
$mail->addAddress('Mailto');
$mail->Subject	= 'subject'; 
$mail->Body		= 'mail Message'; 

if (!$mail->send())
{
	echo 	=  $mail->ErrorInfo;
}
else
{
	echo 'Mail is Sending ok';
}
```

**Comment by Synchro** [comment, marked as answer]
What happened when you followed the link to the troubleshooting guide and did what it said?

Also, why are you disabling certificate verification? It's very unlikely you need to do that; your OS is probably outdated.

**Comment by chris1278** [comment]
So the emails are sent. No matter whether ssl is true or false. That's no problem. And the link only leads to a general description and is not really helpful if the problem is not specifically stated.

Normally the messages that are also contained in the language files should be output, but that doesn't happen.

And the system itself is not outdated. I use the current version of phpmailer. The whole thing is just configured as an extension for the phpbb forum. Even if I put it in a separate html/php form, only the same expression comes up if there is an error.

**Reply by Synchro** [reply]
If your certificates are failing verification, the troubleshooting guide gives clear instructions for how to tell why that is, but by far the most common reason is because your CA certificate bundle is outdated, which is also discussed in-depth in the guide, and has nothing to do with what version of PHPMailer you are using.

One of the very first things the guide suggests is to enable debug output, but you've not done that. Because you may be having a connection-level issue (we can't tell from what you've posted), set `$mail->SMTPDebug = 3;` and show us the output.

**Reply by chris1278** [reply]
This is not about a certificate error. As already mentioned, sending with the SMTP server works. I installed an error because of the error message extra password or smtp host so that I can test this. and it just spits out this troubleshooting page but doesn't specify what exactly caused the error. With the debug it shows something, which is also fine, but it would be good if it also outputs a message when sending if, for example, the password or SMTP host does not match. Or if the email cannot be sent due to an incorrect email address. Especially with regard to valid mx records.

**Reply by Synchro** [reply]
It's true that error messages are not especially fine-grained, however, [this specific error occurs immediately after trying to connect](https://github.com/PHPMailer/PHPMailer/blob/master/src/PHPMailer.php#L2020), so invalid addresses etc have not had an opportunity to be discovered. If errors occur at later stages, you get different error messages, as you can see if you look through the rest of that method. You may also find that you get slightly better feedback if you make use of PHPMailer's exceptions rather than `ErrorInfo`, which only stores the final outcome rather than any intermediate issue.
